stellar-multitenant/pkg/tenant/manager_test.go

```diff
@@ -570,29 +570,86 @@ func Test_Manager_SoftDeleteTenantByID(t *testing.T) {
+		_, err = m.SoftDeleteTenantByID(ctx, tnt.ID)
 		require.Error(t, err)
 		assert.ErrorIs(t, err, ErrTenantDoesNotExist)
 	})
 
 	t.Run("successfully soft deletes tenant", func(t *testing.T) {
 		deactivateTenant(t, ctx, m, tnt)
-		dbTnt, err := m.SoftDeleteTenantByID(ctx, tnt.ID)
-		require.NoError(t, err)
+		dbTnt, dbErr := m.SoftDeleteTenantByID(ctx, tnt.ID)
+		require.NoError(t, dbErr)
 		require.NotNil(t, dbTnt.DeletedAt)
 
-		dbTnt, err = m.GetTenant(ctx, &QueryParams{Filters: map[FilterKey]interface{}{FilterKeyID: tnt.ID}})
-		require.NoError(t, err)
+		dbTnt, dbErr = m.GetTenant(ctx, &QueryParams{Filters: map[FilterKey]interface{}{FilterKeyID: tnt.ID}})
+		require.NoError(t, dbErr)
 		assert.NotNil(t, dbTnt.DeletedAt)
 	})
 }
 
+func TestManager_DeactivateTenantDistributionAccount(t *testing.T) {
+	dbt := dbtest.OpenWithAdminMigrationsOnly(t)
+	defer dbt.Close()
+
+	dbConnectionPool, err := db.OpenDBConnectionPool(dbt.DSN)
+	require.NoError(t, err)
+	defer dbConnectionPool.Close()
+
+	ctx := context.Background()
+
+	m := NewManager(WithDatabase(dbConnectionPool))
+	tnt, err := m.AddTenant(ctx, "myorg1")
+	require.NoError(t, err)
+
+	t.Run("does not deactivate distribution account if not managed by Circle", func(t *testing.T) {
+		err = m.DeactivateTenantDistributionAccount(ctx, tnt.ID)
+		require.NoError(t, err)
+
+		dbTnt, dbErr := m.GetTenant(ctx, &QueryParams{Filters: map[FilterKey]interface{}{FilterKeyID: tnt.ID}})
+		require.NoError(t, dbErr)
+		assert.NotEqual(t, schema.AccountStatusPendingUserActivation, dbTnt.DistributionAccountStatus)
+	})
+
+	t.Run("does not deactivate distribution account if tenant is deactivated", func(t *testing.T) {
+		tnt, err = m.UpdateTenantConfig(
+			ctx, &TenantUpdate{
+				ID:                      tnt.ID,
+				DistributionAccountType: schema.DistributionAccountCircleDBVault,
+				Status:                  pointerTo(DeactivatedTenantStatus),
+			})
+		require.NoError(t, err)
+
+		err = m.DeactivateTenantDistributionAccount(ctx, tnt.ID)
+		require.NoError(t, err)
+
+		dbTnt, dbErr := m.GetTenant(ctx, &QueryParams{Filters: map[FilterKey]interface{}{FilterKeyID: tnt.ID}})
+		require.NoError(t, dbErr)
+		assert.NotEqual(t, schema.AccountStatusPendingUserActivation, dbTnt.DistributionAccountStatus)
+	})
+
+	t.Run("successfully deactivates tenant distribution account", func(t *testing.T) {
+		tnt, err = m.UpdateTenantConfig(
+			ctx, &TenantUpdate{
+				ID:                      tnt.ID,
+				DistributionAccountType: schema.DistributionAccountCircleDBVault,
+				Status:                  pointerTo(ActivatedTenantStatus),
+			})
+		require.NoError(t, err)
+		err = m.DeactivateTenantDistributionAccount(ctx, tnt.ID)
+		require.NoError(t, err)
+
+		dbTnt, dbErr := m.GetTenant(ctx, &QueryParams{Filters: map[FilterKey]interface{}{FilterKeyID: tnt.ID}})
+		require.NoError(t, dbErr)
+		assert.Equal(t, schema.AccountStatusPendingUserActivation, dbTnt.DistributionAccountStatus)
+	})
```

Comment:
👋  could you include a test case to validate that this method won't throw an error if the distribution account status is already `PENDING_USER_ACTIVATION?